engine/server/application/LoginServer/src/shared/SessionApiClient.cpp

```diff
@@ -120,7 +120,7 @@ void SessionApiClient::OnSessionValidate(const apiTrackingNumber trackingNumber,
 	else
 	{
 		// attempt to validate a CS Tool associated with this request.
-		CSToolConnection::validateCSTool( ( uint32 )userData, result, session );
+		CSToolConnection::validateCSTool( ( uint64 )userData, result, session );
```

Comment:
This likely needs updating a bit deeper.  It's being cast as a uint64 because the calling method has userData as a `void *` type.  Because of 64 bits, a `void *` becomes a 64 bit value.  The calling method needs to be updated to not accept a void if at all possible and all callers of the calling method should also be updated.  Great catch!